### DISCUSSION TITLE
grain size

### DISCUSSION THREAD

**Discussion by kittygg** [neperfepx/neper#942] [Q&A] [answered]
How to decrease the small edge size as much as possible while ensuring the 3D grain size is relatively uniform?

**Comment by rquey** [comment]
I'm not sure I understand what you mean. Could you elaborate?

**Reply by kittygg** [reply]
When 3D grains of relatively uniform size are divided using tetrahedron elements, a few elements have a large aspect ratio.

**Reply by rquey** [reply, marked as answer]
Elements with a large aspect ratio are caused by "small edges". Use `-reg 1` to remove them.

If this does not answer your question, please be more specific and provide an example.

**Reply by kittygg** [reply]
neper -T -n  100  -dim 3  -reg 1  -fmax 10   -domain "cube(1,1,1)"  -morpho "gg,size:0.01,1-sphericity:lognormal(0.1,0.03)" -o 3D
How to generate higher quality grains for meshing?

**Reply by rquey** [reply]
`-morpho "gg,size:0.01,1-sphericity:lognormal(0.1,0.03)"` looks wrong, as you are imposing size and sphericity distributions via  `gg` and also via `size:0.01,1-sphericity:lognormal(0.1,0.03)"`. Choose one or the other.

Otherwise, you have `-reg 1`, which is a good thing. You could "relax" `-fmax` (use a higher value) to get more regularization, which will in turn improve element quality. You could finally add `-meshqualmin 1` to get the absolute best element quality, but the improvement will be small.

**Reply by kittygg** [reply]
Thank you, I will try again.
